ReactAndroid/src/main/java/com/facebook/react/uimanager/ViewManagersPropertyCache.java

```diff
@@ -184,11 +193,131 @@ public DoublePropSetter(ReactPropGroup prop, Method setter, int index, double de
     }
 
     @Override
-    protected Object getValueOrDefault(Object value) {
+    protected Object getValueOrDefault(Object value, Context context) {
       return value == null ? mDefaultValue : (Double) value;
     }
   }
 
+  private static class ColorPropSetter extends PropSetter {
+    private static final String JSON_KEY = "resource_paths";
+    private static final String PREFIX_RESOURCE = "@";
+    private static final String PREFIX_ATTR = "?";
+    private static final String PACKAGE_DELIMITER = ":";
+    private static final String PATH_DELIMITER = "/";
+    private static final String ATTR = "attr";
+    private static final String ATTR_SEGMENT = "attr/";
+
+    private final int mDefaultValue;
+
+    public ColorPropSetter(ReactProp prop, Method setter) {
+      super(prop, "mixed", setter);
+      mDefaultValue = 0;
+    }
+
+    public ColorPropSetter(ReactProp prop, Method setter, int defaultValue) {
+      super(prop, "mixed", setter);
+      mDefaultValue = defaultValue;
+    }
+
+    @Override
+    protected Object getValueOrDefault(Object value, Context context) {
+      if (value == null) {
+        return mDefaultValue;
+      }
+
+      if (value instanceof Double) {
+        return ((Double) value).intValue();
+      }
+
+      if (context == null) {
+        throw new RuntimeException("Context may not be null.");
+      }
+
+      if (value.getClass() == ReadableMap.class || value.getClass() == ReadableNativeMap.class) {
+        ReadableMap map = (ReadableMap) value;
+        ReadableArray resourcePaths = map.getArray(JSON_KEY);
+
+        if (resourcePaths == null) {
+          throw new JSApplicationCausedNativeException("ColorValue: The `" + JSON_KEY + "` must be an array of color resource path strings.");
+        }
+
+        for (int i = 0; i < resourcePaths.size(); i++) {
+          String resourcePath = resourcePaths.getString(i);
+
+          if (resourcePath == null || resourcePath.isEmpty()) {
+            continue;
+          }
+
+          boolean isResource = resourcePath.startsWith(PREFIX_RESOURCE);
+          boolean isThemeAttribute = resourcePath.startsWith(PREFIX_ATTR);
+
+          resourcePath = resourcePath.substring(1);
+
+          try {
+            if (isResource) {
+              return resolveResource(context, resourcePath);
+            } else if (isThemeAttribute) {
+              return resolveThemeAttribute(context, resourcePath);
+            }
+          } catch (Resources.NotFoundException exception) {
+            exception.printStackTrace();
```

Comment:
It cannot throw.   If the resource could not be found it purposely does nothing to allow the for loop to continue and try the next fallback resource in the array.  If none of the fallbacks are found then the exception immediately after the for loop will be thrown.

I'll add this comment to the code.

I'm not sure if we want to log this as it could flood the log?   The purpose is to allow the latest OS hotness to be early in the array and older fallbacks to be later in the array.